schema_based_on_diagram.sql

```diff
@@ -0,0 +1,49 @@
+//Database schema based on a given diagram
+
+CREATE TABLE patients (
+    id serial primary key NOT NULL,
+    name varchar,
+    date_of_birth date
+);
+
+CREATE TABLE medical_histories (
+    id serial primary key NOT NULL,
+    admitted_at timestamp,
+    patient_id int,
+    status varchar ,
+    PRIMARY KEY(id),
+    CONSTRAINT fk_patient,
+      FOREIGN KEY(id) ,
+	  REFERENCES patient(patient_id)
+);
+
+CREATE TABLE treatments (
+    id serial primary key NOT NULL,
+    type varchar,
+    name varchar 
+    CONSTRAINT fk_patient
+    FOREIGN KEY(id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoices (
+    id serial primary key NOT NULL,
+    total_amount decimal,
+    generated_at timestamp,
+    payed_at timestamp,
+    medical_history_id int
+    date_of_birth date 
+    CONSTRAINT fk_invoice,
+    FOREIGN KEY(id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+    id serial primary key NOT NULL,
+    unit_price decimal,
+    quantity int,
+    total_price decimal,
+    invoice_id int,
+    treatment_id int 
+    CONSTRAINT fk_invoice
+    FOREIGN KEY(id) REFERENCES invoice_id(id)
+    FOREIGN KEY(id) REFERENCES treatments(id)
+);
```

Comment:
- [ ]  As it is written in the project requirements, you are required to write the indexes for all your foreign keys. As you will be having 6 tables, this means that you also need 6 indexes as well for each of the tables. Kindly include them, **Example:** ```CREATE INDEX ON medical_histories (patient_id);```

![image](https://user-images.githubusercontent.com/86318284/207371666-35c81594-e660-44ef-b8de-4def096b39ad.png)